thefuck/rules/git_branch_0v_to_dash_v.py

```diff
@@ -0,0 +1,13 @@
+from thefuck.shells import shell
+from thefuck.specific.git import git_support
+
+
+@git_support
+def match(command):
+    return (command.script_parts
+            and command.script_parts[1:] == 'branch 0v'.split())
```

Comment:
No need to split. Could use a list instead.
```suggestion
            and command.script_parts[1:] == ['branch', '0v'])
```

Not that the way I use Git is of any importance, but I seldom use `-v`, if ever. I use `-r` or `-l` or `-d` much more frequently. So perhaps the matching should be more comprehensive. How about the following:

```python
if len(a) == 2 and a.startswith("0")
```

Then the rule could have a function to return the first occurrence of that in `script_parts`. Something like:

```python
    return next((a for a in script_parts if len(a) == 2 and a.startswith("0")), "")
```

That function could be [*memoized*](https://github.com/nvbn/thefuck/blob/master/thefuck/rules/git_add.py#L8) and used in `get_new_command` as well.

What do you think?

Reply:
> What are the advantages of making the code change on line 8?

The result of `'branch 0v'.split()` is exactly `['branch', '0v']`. In using the actual list, you're saving a few steps ([i.e.](https://www.merriam-webster.com/video/ie-vs-eg) calling a function that would split the string and return a list).

> Just looking at the code here on the PR and not in my editor, what I _think_ you're saying is that we can use a list comprehension to build a list of actions to take based on what the character after the erroneous `0`.

Yes. More precisely, that is a generator expression. It's better to use them in such cases as it stops at the first element found. The action would be the same for any of the characters after `0`, though: replace `0` with `-`.

> [...] Then I think you're saying that I can use the memoize annotation (is that what they're called in Python) to return the pre-defined "correct" and corrective action to take, thus allowing this functionality to be expanded upon bit by bit in future PRs.

You're partially correct regarding the memoization. There's a pattern for it in the project, as you can see in the link I posted. Let's take it as an example:

https://github.com/nvbn/thefuck/blob/365db1ee41d7bcd9f62d910f3b20361ea3499d92/thefuck/rules/git_add.py#L5-L27

Note that the `_get_missing_file` function is used in both `match` (on line 20) and `get_new_command` (on line 25). These are two separate *“phases”* of the *“correction”* process. TheFuck first tries to match a rule. If that rule matches, it then gets the new command. That means `_get_missing_file` is called two times. That's when the memoization comes in handy.

ps.: thinking over it again, maybe that's also a "Nit" if you consider the fact that the proposed function wouldn't be much expensive.

Reply:
I took the ideas presented here and made a run at implementation. Ultimately, I chose to use a dictionary with the flag as a key and a tuple of the "undo" plus the "what you meant to".

Re-requesting review; hopefully these changes will, at long last, allow this PR to get merged.